htmldocx/h2d.py

```diff
@@ -267,21 +273,29 @@ def parse_dict_string(self, string, separator=';'):
         return string_dict
 
     def handle_li(self):
+        restart_numbering = False
         # check list stack to determine style and depth
         list_depth = len(self.tags['list'])
         if list_depth:
             list_type = self.tags['list'][-1]
         else:
-            list_type = 'ul' # assign unordered if no tag
+            list_type = 'ul'  # assign unordered if no tag
 
         if list_type == 'ol':
             list_style = self.ol_style
+
+            # Restart numbering if previous paragraph style is not the same as self.ol_style
+            previous_paragraph = self.doc.paragraphs[-1]
+            if previous_paragraph.style.name != self.ol_style:
+                restart_numbering = True
         else:
             list_style = self.ul_style
 
-        self.paragraph = self.doc.add_paragraph(style=list_style)            
+        self.paragraph = self.doc.add_paragraph(style=list_style)
         self.paragraph.paragraph_format.left_indent = Inches(min(list_depth * LIST_INDENT, MAX_INDENT))
         self.paragraph.paragraph_format.line_spacing = 1
+        if restart_numbering:
+            self.doc.restart_numbering(self.paragraph)
```

Comment:
I think this is wrong. Need to find out of self.doc is analagous to `BaseDocument`. Also probably need to check if self.doc is None. It's only set in `set_initial_attrs` and it appears to be optional there.